### DISCUSSION TITLE
Reducing license file size for air-gapped activation

### DISCUSSION THREAD

**Discussion by JonHollander** [keygen-sh/community#118] [Q&A] [answered]
Hi-
I'm implementing an air-gapped activation system for photobooth software. Because the client systems all have cameras, I would like to encode the entire license file in a QR code (or similar), so that user's don't need to use USB drives to transfer files. Unfortunately, the current license files are around 3x too large to be packed into a single QR code. 

The decoded license data has a lot of information that I don't need because I'm using machine fingerprinting. Is there any way to reduce the license payload size? 

I looked at the `?include` query parameter for the checkout action. But calling this with different values or skipping it entirely doesn't seem to have any effect on the returned data; the `included` field seems to include everything regardless.

**Comment by ezekg** [comment, marked as answer]
There is currently no way to reduce the size of a license file. By design, it includes the entire license object, plus any included relationships (which are optional), using our standard API object format. This can end up being a lot of data, and we have no current plans to make these smaller at this time. I'd suggest distributing the license files to the offline devices another way, such as via email, bluetooth, or USB.

One alternative would be [signed license keys](https://keygen.sh/docs/api/cryptography/#cryptographic-keys), but those come with their own caveats (immutability, lack of encryption, etc.)

**Reply by JonHollander** [reply]
Thanks! I'll take a look at signed license keys.
